solidity-mode.el

```diff
@@ -551,14 +551,16 @@ Cursor must be at the function's name.  Does not currently work for constructors
   (let* ((spacetabs "[\t\n ]+")
          (optional-spacetabs "[\t\n ]*")
          (ident-group "\\([A-Za-z_][A-Za-z0-9_]*\\)")
-         (ctr-ident-group "\\(constructor\\)")
+         (constructor-ident-group "\\(constructor\\)")
+         (fallback-ident-group "\\(fallback\\)")
          (modifier (mapconcat 'identity
                               '("payable" "public" "private" "external" "internal" "view" "pure")
                               "\\|"))
          (modifiers (concat "\\(?:\\(?:" modifier "\\)" spacetabs "\\)*")))
     `(("function", (concat "^" optional-spacetabs "function" spacetabs ident-group) 1)
       ("modifier", (concat "^" optional-spacetabs "modifier" spacetabs ident-group) 1)
-      ("constructor", (concat "^" optional-spacetabs ctr-ident-group) 1)
+      ("constructor", (concat "^" optional-spacetabs constructor-ident-group) 1)
+      ("function", (concat "^" optional-spacetabs fallback-ident-group) 1)
```

Comment:
Should this be fallback?
```suggestion
      ("fallback", (concat "^" optional-spacetabs fallback-ident-group) 1)
```

Reply:
The first item of this list is the item type. For example, here are the item types that are supported by `helm-imenu`

https://github.com/emacs-helm/helm/blob/1697afdfdc00e99df7dc910eebf3d0da73c4f5ae/helm-imenu.el#L107-L178

Since `fallback` is a function, I think it's better to return the `function` type to make it compatible with other imenu libraries.